servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/TcpClientChannelInitializer.java

```diff
@@ -94,8 +94,7 @@ public TcpClientChannelInitializer(final ReadOnlyTcpClientConfig config,
                     channel -> new TcpConnectionInfo(channel,
                             // ExecutionContext can be null if users used deprecated ctor
                             executionContext == null ? null : channelExecutionContext(channel, executionContext),
-                            sslConfig, config.idleTimeoutMs()),
-                    sslConfig != null && !deferSslHandler, true));
+                            sslConfig, config.idleTimeoutMs()), true, !deferSslHandler ? sslConfig : null));
```

Comment:
To avoid the boolean flip:
```suggestion
                            sslConfig, config.idleTimeoutMs()), true, deferSslHandler ? null : sslConfig));
```